### PR TITLE
TDR-2135 Additional metadata start page

### PR DESCRIPTION
**Pull request by MancunianSam**
* Add NHS frontend package. This will be added in another PR but I'll add this in here to keep things working.
* Add a route. I've also set a message for the progress indicator. This is likely to change but this will do as a placeholder.
* Add controller and tests. For now, if there's no parent folder, they'll just get the error page. The logic around what to do based on the consignment status is already pretty complicated and I didn't want to make it worse on a page that's subject to change.

### CODE REVIEW

**Review by TomJKing — COMMENTED** [review]
Looks good. Some minor comments

File: app/controllers/AdditionalMetadataController.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import scala.concurrent.Future
+
+class AdditionalMetadataController @Inject () (val consignmentService: ConsignmentService,
+                                               val keycloakConfiguration: KeycloakConfiguration,
+                                               val controllerComponents: SecurityComponents
+                                              ) extends TokenSecurity {
+
+  def start(consignmentId: UUID): Action[AnyContent] = standardTypeAction(consignmentId) { implicit request: Request[AnyContent] =>
+    for {
+      reference <- consignmentService.getConsignmentRef(consignmentId, request.token.bearerAccessToken)
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply app/controllers/AdditionalMetadataController.scala:20]: Ah good spot. I've updated it to use the new endpoint and changed the tests.

File: app/views/standard/additionalMetadataStart.scala.html [hunk 2]
```diff
@@ -0,0 +1,37 @@
+@import views.html.partials.{transferReference, progressIndicator}
+@import java.util.UUID
+
+@(parentFolder: String, consignmentRef: String, consignmentId: UUID, name: String)(implicit messages: Messages, request: RequestHeader)
+@main(title = "Add, edit or delete metadata", name = name) {
+  @defining(play.core.PlayVersion.current) { version =>
+    @progressIndicator(Messages("additionalMetadataStart.progress"))
+    <div class="govuk-grid-row additional-metadata-start">
+      <div class="govuk-grid-column-two-thirds">
+        <h1 class="govuk-heading-l">Add, edit or delete metadata</h1><h1 class="govuk-heading-s">Folder uploaded: @parentFolder</h1>
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply app/views/standard/additionalMetadataStart.scala.html:10]: I've changed it. Bad copying and pasting as ever.

File: test/controllers/AdditionalMetadataControllerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+  override def afterEach(): Unit = {
+    wiremockServer.resetAll()
+    wiremockServer.stop()
+  }
+
+  val checkPageForStaticElements = new CheckPageForStaticElements
+  implicit val ec: ExecutionContext = ExecutionContext.global
+
+  "AdditionalMetadataController" should {
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply test/controllers/AdditionalMetadataControllerSpec.scala:36]: The API will return an error if this happens so I've put a test in for dealing with that error.

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**